pyInterface/test_MC.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+### BEGIN CONFIGURATION ###
+
+	### BEGIN PHASE SPACE ###
+
+		# mass bin (MeV)
+		MASS=1800
+		BINWIDTH=20
+
+		# number of phase space events to generate 
+		NMB_PS_EVENTS=500	
+
+	### END PHASE SPACE ###
+
+	### BEGIN PARTICLE DATA ###
+
+		# particle data table
+		PARTICLE_DATA_TABLE="${ROOTPWA}/particleData/particleDataTable.txt"
+
+	### END PARTICLE DATA ###
+
+	### BEGIN PSEUDO DATA ###
+
+		# number of weighted pseudo data events to generate
+		NMB_PSEUDO_EVENTS=500
+
+	### END PSEUDO DATA ###
+
+
+#-- END CONFIGURATION --#
+
+
+### BEGIN PREPARATION ###
+
+if [ -d "${ROOTPWA}" ]; then
+	echo "RootPwa installation found in: ${ROOTPWA}"
+	echo "Starting MC test ..."
+else
+	echo "Error no installation of RootPwa found! Aborting."
+	exit 1
+fi
+
+TESTDIR=${PWD}/ROOTPWA_MC_TEST
+
+if [ -d "${TESTDIR}" ]; then
+	echo "Test directory ${TESTDIR} already exists! Aborting."
+	exit 1
+fi
+
+echo "Creating directories ..."
+
+mkdir "${TESTDIR}"
+mkdir "${TESTDIR}/data"
+mkdir "${TESTDIR}/weighted_mc_data"
+mkdir "${TESTDIR}/amps"
+mkdir "${TESTDIR}/ints"
+mkdir "${TESTDIR}/keyfiles"
+mkdir "${TESTDIR}/reference_fit"
+mkdir "${TESTDIR}/fits"
+mkdir "${TESTDIR}/log"
+
+echo "Test directory and subfolders created."
+echo "Changing to test directory."
+
+cd "${TESTDIR}"
+
+exec > >(tee -i ./log/logfile.txt)
+exec 2>&1
+
+echo "Copying necessary files to test directory ..."
+
+cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/generator_noBeamSimulation.conf" "./"
+cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/GenerateKeyfiles.sh" "./"
+cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/template.key" "./"
+cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/wavelist.compass.2008.88waves" "./"
+cp "${ROOTPWA}/pyInterface/rootpwa.config" "./"
+cp "${ROOTPWA}/pyInterface/mcTest/reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root.example" "./reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root"
+
+#-- END PREPARATION --#
+
+### BEGIN MONTE CARLO GENERATION ###
+
+# generate phase space data
+echo "Generating phase space ..."
+${ROOTPWA}/pyInterface/genpw.py -n $NMB_PS_EVENTS -p "${PARTICLE_DATA_TABLE}" -M $MASS -B $BINWIDTH "./generator_noBeamSimulation.conf" -o "./data/phase_space_MASS_${MASS}-$((MASS+BINWIDTH))_N_${NMB_PS_EVENTS}.root"
+
+# generate the keyfiles
+echo "Generating keyfiles ..."
+DESTINATION_DIR="${TESTDIR}/keyfiles"
+export PARTICLE_DATA_TABLE
+export DESTINATION_DIR
+./GenerateKeyfiles.sh 
```

Comment:
by also setting the `WAVESET_FILES` variable to `"${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/wavelist.compass.2008.88waves"` the number of created keyfiles could be reduced. Only those amplitudes actually used in the fit later would have to be calculated. (don't know whether this is really required.)